spec/unit/copy_service_spec.rb

```diff
@@ -31,4 +33,48 @@
       described_class.run(opts, ['service_id'])
     end
   end
+
+  context '#copy_service_params' do
+    let(:params) do
+      %w[
+        name backend_version deployment_option description
+        system_name end_user_registration_required
+        support_email tech_support_email admin_support_email
+      ]
+    end
+
+    let(:source_service_params) do
+      params.each_with_object({}) do |key, target|
+        target[key] = random_lowercase_name
+      end
+    end
+
+    it 'all expected params are copied' do
+      new_service_params = described_class.copy_service_params(source_service_params, nil)
+
+      expect(new_service_params).to include(*params)
+    end
+
+    it 'extra params are not copied' do
```

Comment:
Why we don't want to copy all params?

Reply:
trying to avoid sending unexpected params to 3scale API. Just for sanity.

Reply:
this is the response body from `GET /admin/api/services/#{id}`
```
{
  "id": 2555417761220,
  "name": "some_name_1",
  "state": "incomplete",
  "system_name": "some_name_1",
  "end_user_registration_required": true,
  "backend_version": "2",
  "deployment_option": "hosted",
  "support_email": "thomas@webtypes.com",
  "description": "some description",
  "created_at": "2018-10-17T11:12:46+02:00",
  "updated_at": "2018-10-17T11:12:46+02:00",
  "links": [
    {
      "rel": "metrics",
      "href": "https://3scale-supertest-admin.3scale.net/admin/api/services/2555417761220/metrics"
    },
    {
      "rel": "end_user_plans",
      "href": "https://3scale-supertest-admin.3scale.net/admin/api/services/2555417761220/end_user_plans"
    },
    {
      "rel": "self",
      "href": "https://3scale-supertest-admin.3scale.net/admin/api/services/2555417761220"
    },
    {
      "rel": "service_plans",
      "href": "https://3scale-supertest-admin.3scale.net/admin/api/services/2555417761220/service_plans"
    },
    {
      "rel": "application_plans",
      "href": "https://3scale-supertest-admin.3scale.net/admin/api/services/2555417761220/application_plans"
    },
    {
      "rel": "features",
      "href": "https://3scale-supertest-admin.3scale.net/admin/api/services/2555417761220/features"
    }
  ]
}
```

3scale API to create service `POST /admin/api/services.xml`

does not need many of them. 

Instead you would allow them all to be included in create service or update service request body?

Reply:
Hmmm. I'm not sure if the API is going to just strip unacceptable parameters or throw an error.
If it would strip them, then we are free to send it whole back. Then there would be no need for maintenance when we expose more fields. Or going for a blacklist instead of a whitelist.

But anyway, it is good as it is.

Reply:
the idea was whitelist as it is :+1: